asio/include/asio/ssl/impl/context.ipp

```diff
@@ -922,6 +928,69 @@ ASIO_SYNC_OP_VOID context::do_use_tmp_dh(
   ASIO_SYNC_OP_VOID_RETURN(ec);
 }
 
+void context::use_tmp_ecdh_file(const std::string& certificate)
+{
+  asio::error_code ec;
+  use_tmp_ecdh_file(certificate, ec);
+  asio::detail::throw_error(ec, "use_tmp_ecdh_file");
+}
+
+ASIO_SYNC_OP_VOID context::use_tmp_ecdh_file(const std::string& certificate,
+        asio::error_code& ec)
+{
+  ::ERR_clear_error();
+
+  bio_cleanup bio = { ::BIO_new_file(certificate.c_str(), "r") };
+  if (bio.p)
+  {
+    return do_use_tmp_ecdh(bio.p,ec);
+  }
+
+  ec = asio::error_code(
+      static_cast<int>(::ERR_get_error()),
+      asio::error::get_ssl_category());
+  ASIO_SYNC_OP_VOID_RETURN(ec);
+}
+
+ASIO_SYNC_OP_VOID context::do_use_tmp_ecdh(
+        BIO* bio, asio::error_code& ec)
+{
+  ::ERR_clear_error();
+
+  int nid = NID_undef;
+
+  x509_cleanup x509 = { ::PEM_read_bio_X509(bio, NULL, 0, NULL) };
+  if (x509.p)
+  {
+    evp_pkey_cleanup pkey = { ::X509_get_pubkey(x509.p) };
+    if(pkey.p)
+    {
+      int type = EVP_PKEY_type(pkey.p->type);
```

Comment:
In order to compile with OpenSSL 1.1, as well as 1.0, there's a small change you can make here. See https://github.com/sony/nmos-cpp/commit/991630e5818b9b5965494711cba4eb001eeec942.

Reply:
Done. Thanks for the link.